dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -480,44 +434,106 @@ public Response getActive(@Context HttpServletRequest httpServletRequest, @Conte
         try {
-            InitDataObject init = auth(params,httpServletRequest, httpServletResponse);
+            InitDataObject init = auth(httpServletRequest, httpServletResponse);
 
             //Creating an utility response object
-            ResourceResponse responseResource = new ResourceResponse( init.getParamsMap() );
 
-            return responseResource.response( new Gson().toJson( APILocator.getContentletIndexAPI().listDotCMSIndices() ) );
+            return Response.ok(new ResponseEntityView(APILocator.getContentletIndexAPI().listDotCMSIndices())).build();
         } catch (Exception e) {
             Logger.error(this.getClass(),"Exception trying to list indices: " + e.getMessage(), e);
             return ResponseUtil.mapExceptionResponse(e);
         }
     }
+    
+    
+    /**
+     * sends an admin growl message
+     * @param message
+     * @param severity
+     * @param user
+     * @param millis
+     */
+    @CloseDBIfOpened
+    private void sendAdminMessage(final String message, final MessageSeverity severity, final User user, final long millis) {
+
+
+        final SystemMessageBuilder systemMessageBuilder = new SystemMessageBuilder();
+
+
+
+        SystemMessage systemMessage = systemMessageBuilder.setMessage(message).setType(MessageType.SIMPLE_MESSAGE)
+                        .setSeverity(severity).setLife(millis).create();
+
+        SystemMessageEventUtil.getInstance().pushMessage(systemMessage, ImmutableList.of(user.getUserId()));
+    }
+    
+    @CloseDBIfOpened
+    @PUT
+    @JSONP
+    @NoCache
+    @Path("/{indexName: .*}")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response modIndex(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                    @PathParam("indexName") final String indexName, @QueryParam("action") final String action) throws DotDataException, IOException {
+
+        final InitDataObject init = auth(request, response);
+        final IndexAction indexAction = IndexAction.fromString(action);
+        
+        if(indexExists(indexName) ){
+            return Response.status(404).build();
+        }
+        
+        
+        switch(indexAction){
+            case DEACTIVATE:
+                APILocator.getContentletIndexAPI().deactivateIndex(indexName);
+                break;
+            case CLEAR:
+                APILocator.getESIndexAPI().clearIndex(indexName);
+                break;
+            case OPEN:
+                APILocator.getESIndexAPI().openIndex(indexName);
+                break;
+            case CLOSE:
+                APILocator.getESIndexAPI().closeIndex(indexName);
+                break;
+            default:
+                APILocator.getContentletIndexAPI().activateIndex(indexName);
+            
+        }
+        String message = indexAction.name().toLowerCase() + " " + indexName;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'message' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)